lit_gpt/lora.py

```diff
@@ -87,6 +88,71 @@ def __init__(self, r: int, lora_alpha: int, lora_dropout: float, merge_weights:
         self.merge_weights = merge_weights
 
 
+class Linear(nn.Linear, LoRALayer):
+    # LoRA implemented in a dense layer
+    def __init__(
+        self,
+        in_features: int,
+        out_features: int,
+        r: int = 0,
+        lora_alpha: int = 1,
+        lora_dropout: float = 0.,
+        fan_in_fan_out: bool = False, # Set this to True if the layer to replace stores weight like (fan_in, fan_out)
+        merge_weights: bool = True,
+        **kwargs
+    ):
+        nn.Linear.__init__(self, in_features, out_features, **kwargs)
+        LoRALayer.__init__(self, r=r, lora_alpha=lora_alpha, lora_dropout=lora_dropout,
+                           merge_weights=merge_weights)
+
+        self.fan_in_fan_out = fan_in_fan_out
+        # Actual trainable parameters
+        if r > 0:
+            self.lora_A = nn.Parameter(self.weight.new_zeros((r, in_features)))
+            self.lora_B = nn.Parameter(self.weight.new_zeros((out_features, r)))
+            self.scaling = self.lora_alpha / self.r
+            # Freezing the pre-trained weight matrix
+            self.weight.requires_grad = False
+        self.reset_parameters()
+        if fan_in_fan_out:
+            self.weight.data = self.weight.data.transpose(0, 1)
+
+    def reset_parameters(self):
+        nn.Linear.reset_parameters(self)
```

Comment:
```suggestion
        super().reset_parameters()
```

Reply:
This should work since `nn.Linear` is listed first (in the list of parent classes). 
But maybe explicitly calling is not such a bad thing?

Reply:
I'm not aware of the whole situation, just noticed from latest PRs that bnb (I assume it's bits_and_bytes) does some monkey patching with linear layer.
The question is: this should also affect ~MergedLinear~ LoRAQKVLinear too, right?